### PR TITLE
Transaction serializability

### PR DESCRIPTION
**Pull request by mudit1102**

### CODE REVIEW

File: src/main/java/com/work/management/service/employee/impl/EmployeeServiceImpl.java [hunk 1]
```diff
@@ -54,6 +57,7 @@ public EmployeeDto getEmployeeByUserName(String username) {
   }
 
   @Override
+  @Transactional(isolation = Isolation.SERIALIZABLE, propagation = Propagation.REQUIRES_NEW)
   public EmployeeDto updateEmployeeEntity(EmployeeDto employeeDto) {
```
Comment (ankitmathur3193) [inline src/main/java/com/work/management/service/employee/impl/EmployeeServiceImpl.java:61]: I think you need to change the propagation level

Comment (mudit1102) [reply src/main/java/com/work/management/service/employee/impl/EmployeeServiceImpl.java:61]: Done

File: src/main/java/com/work/management/service/employee/impl/EmployeeServiceImpl.java [hunk 2]
```diff
@@ -11,9 +11,12 @@
 import org.springframework.beans.BeanUtils;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
 
 @Service
-final class EmployeeServiceImpl implements EmployeeService {
+public class EmployeeServiceImpl implements EmployeeService {
```
Comment (ankitmathur3193) [inline src/main/java/com/work/management/service/employee/impl/EmployeeServiceImpl.java:20]: Is removing final and making it public neccessary...try testing with all combination..like removing public...and removing public and making it final...just see..if it is working or not?

**Review by ankitmathur3193 — DISMISSED** [review]

File: src/main/java/com/work/management/service/employee/impl/EmployeeServiceImpl.java (hunk 2 above)
Comment (mudit1102) [reply src/main/java/com/work/management/service/employee/impl/EmployeeServiceImpl.java:20]: Done

**Pull request merged**